specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-04-01/machineLearningServices.json

```diff
@@ -4701,6 +4717,260 @@
+          ]
+        }
+      },
+      "additionalProperties": true
+    },
+    "Image": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "default": "docker",
+          "enum": [
+            "docker",
+            "azureml"
+          ],
+          "type": "string",
+          "title": "Type of the image",
+          "description": "Type of the image. Possible values are: docker - For docker images. azureml - For AzureML images",
+          "x-ms-enum": {
+            "name": "ImageType",
+            "modelAsString": true
+          }
+        },
+        "reference": {
+          "type": "string",
+          "description": "Image reference"
+        }
+      },
+      "additionalProperties": true
+    },
+    "EnvironmentVariable": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "enum": [
+            "local"
+          ],
+          "default": "local",
+          "title": "Type of Environment Variable",
+          "description": "Type of the Environment Variable. Possible values are: local - For local variable",
+          "type": "string",
+          "x-ms-enum": {
+            "name": "EnvironmentVariableType",
+            "modelAsString": true
+          }
+        },
+        "value": {
+          "type": "string",
+          "description": "Value of the Environment variable"
+        }
+      },
+      "additionalProperties": true
+    },
+    "Docker": {
+      "type": "object",
+      "properties": {
+        "privileged": {
+          "type": "boolean",
+          "description": "Indicate whether container shall run in privileged or non-privileged mode.",
+          "x-nullable": true
+        }
+      },
+      "additionalProperties": true
+    },
+    "Endpoint": {
+      "type": "object",
+      "properties": {
+        "protocol": {
+          "enum": [
+            "tcp",
+            "udp",
+            "http"
+          ],
+          "default": "tcp",
+          "title": "Endpoint Communication Protocol",
+          "description": "Protocol over which communication will happen over this endpoint",
+          "x-ms-enum": {
+            "name": "protocol",
+            "modelAsString": true
+          },
+          "type": "string"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the Endpoint"
+        },
+        "target": {
+          "format": "int32",
+          "type": "integer",
+          "description": "Application port inside the container."
+        },
+        "published": {
+          "format": "int32",
+          "type": "integer",
+          "description": "Port over which the application is exposed from container.",
+          "x-nullable": true
+        },
+        "hostIp": {
+          "description": "Host IP over which the application is exposed from the container",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "VolumeDefinition": {
```

Comment:
>"VolumeDefinition": { [](http://example.com/codeflow?start=4&length=21)

Please add Title or Description to this definition.

Reply:
Added, Thanks for bringing it up.